pkg/api/product_cache.go

```diff
@@ -59,41 +61,58 @@ func NewProductCache(client HTTPClient, token string, logger log.Logger) *Produc
 
 // Refresh requests data from the Fastly API and stores data in the cache.
 func (p *ProductCache) Refresh(ctx context.Context) error {
-	for _, product := range Products {
-		if product == ProductDefault {
-			continue
-		}
-		uri := fmt.Sprintf("https://api.fastly.com/entitled-products/%s", product)
 
-		req, err := http.NewRequestWithContext(ctx, "GET", uri, nil)
-		if err != nil {
-			return fmt.Errorf("error constructing API product request: %w", err)
-		}
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.fastly.com/entitlements", nil)
+	if err != nil {
+		return fmt.Errorf("error constructing API product request: %w", err)
+	}
 
-		req.Header.Set("Fastly-Key", p.token)
-		req.Header.Set("Accept", "application/json")
-		resp, err := p.client.Do(req)
-		if err != nil {
-			return fmt.Errorf("error executing API product request: %w", err)
-		}
-		defer resp.Body.Close()
+	req.Header.Set("Fastly-Key", p.token)
+	req.Header.Set("Accept", "application/json")
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error executing API product request: %w", err)
+	}
+	defer resp.Body.Close()
 
-		if resp.StatusCode != http.StatusOK {
-			return NewError(resp)
-		}
+	if resp.StatusCode != http.StatusOK {
+		return NewError(resp)
+	}
 
-		var response Product
+	var response entitlementsResponse
 
-		if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
-			return fmt.Errorf("error decoding API product response: %w", err)
-		}
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return fmt.Errorf("error decoding API product response: %w", err)
+	}
+
+	activeProducts := make(map[string]interface{})
 
-		level.Debug(p.logger).Log("product", response.Product.Name, "hasAccess", response.HasAccess)
+	for _, customer := range response.Customers {
+		if customer.Contracts == nil {
```

Comment:
You don't this `nil` check. You can range over a `nil` slice (it behave just like a slice with zero entries). Same for `contract.Items`

Reply:
Got it, thanks! Updated.